ambry-frontend/src/main/java/com.github.ambry.frontend/AmbryBlobStorageService.java

```diff
@@ -1104,44 +1091,43 @@ private void injectAccountAndContainerUsingAccountAndContainerHeaders(RestReques
    * {@link Container} into the {@link RestRequest}.
    * @param blobIdStr The blobId string to get the target {@link Account} and {@link Container} id。
    * @param restRequest The rest request to insert the target {@link Account} and {@link Container}.
-   * @throws RestServiceException if 1) either {@link Account} or {@link Container} could not be identified; or 2)
+   * @throws RestServiceException if 1) either {@link Account} or {@link Container} could not be found; or 2)
    *                              either {@link Account} or {@link Container} were explicitly specified as
    *                              {@link Account#UNKNOWN_ACCOUNT} or {@link Container#UNKNOWN_CONTAINER}.
    */
   private void injectTargetAccountAndContainerFromBlobId(String blobIdStr, RestRequest restRequest)
       throws RestServiceException {
+    BlobId blobId;
     try {
-      BlobId blobId = new BlobId(blobIdStr, clusterMap);
-      Account targetAccount = accountService.getAccountById(blobId.getAccountId());
-      if (targetAccount == null) {
-        frontendMetrics.getHeadDeleteUnrecognizedAccountCount.inc();
-        // @todo The check can be removed once HelixAccountService is running with UNKNOWN_ACCOUNT created.
-        if (blobId.getAccountId() != Account.UNKNOWN_ACCOUNT_ID) {
-          throw new RestServiceException(
-              "Account from blobId=" + blobIdStr + "with accountId=" + blobId.getAccountId() + " cannot be recognized",
-              RestServiceErrorCode.InvalidAccount);
-        } else {
-          logger.debug(
-              "Account cannot be identified for blobId={} with accountId={}. Setting targetAccount to UNKNOWN_ACCOUNT",
-              blobIdStr, blobId.getAccountId());
-          targetAccount = Account.UNKNOWN_ACCOUNT;
-        }
-      }
-      Container targetContainer = targetAccount.getContainerById(blobId.getContainerId());
-      if (targetContainer == null) {
-        frontendMetrics.getHeadDeleteUnrecognizedContainerCount.inc();
-        throw new RestServiceException(
-            "Container from blobId=" + blobIdStr + "with accountId=" + blobId.getAccountId() + " containerId="
-                + blobId.getContainerId() + " cannot be recognized", RestServiceErrorCode.InvalidContainer);
-      }
-      setTargetAccountAndContainerInRestRequest(restRequest, targetAccount, targetContainer);
-    } catch (RestServiceException e) {
-      throw e;
+      blobId = new BlobId(blobIdStr, clusterMap);
     } catch (Exception e) {
       throw new RestServiceException(
           "Invalid blob id=" + blobIdStr + " during injecting account and container from blobId string",
```

Comment:
minor: think you can drop the ".. during injecting ..." and just say invalid blob id and the id